### ISSUE TITLE
Segmentation fault when trying to run on OSX

### ISSUE THREAD

**Issue by aconbere** [binji/binjgb#16]
```bash
[anders/fix_compile_errors_on_osx] % ./bin/binjgb ~/Projects/blarggs-tests/cpu_instrs/source/test.gb
title: ""
cgb flag: CGB_FLAG_SUPPORTED
sgb flag: SGB_FLAG_NONE
cart type: CART_TYPE_MBC1
rom size: ROM_SIZE_32K
ext ram size: EXT_RAM_SIZE_NONE
header checksum: 0x66 [OK]
GL_VERTEX_SHADER ERROR: ERROR: 0:1: '' :  version '130' is not supported
ERROR: 0:2: '' :  #version required and missing.
ERROR: 0:2: 'attribute' : syntax error: syntax error

zsh: segmentation fault  ./bin/binjgb ~/Projects/blarggs-tests/cpu_instrs/source/test.gb
```

It appears to originate in the hard coding of the `GL_VERTEX_SHADER` version: https://github.com/binji/binjgb/blob/master/src/host-ui-imgui.cc#L116

However bumping that to version 150 which I believe is the current version available on my machine moves the error to an attribute syntax error:

```
GL_VERTEX_SHADER ERROR: ERROR: 0:2: 'attribute' : syntax error: syntax error
```

**Comment by binji** [comment]
Does #17 fix this for you?

**Comment by aconbere** [comment]
Ha, I was just testing this morning. I think there's one more bug where `texture2D` is deprecated (now called just `texture`). But with that change it now runs.

Window is a bit oddly sized, but maybe that's expected?

<img width="647" alt="screen shot 2018-11-14 at 9 02 40 am" src="https://user-images.githubusercontent.com/17713/48498910-16771080-e7ec-11e8-9b09-f9e4e464ad06.png">

**Comment by aconbere** [comment]
3a1c729772b48ca44aeaff17090c7ffe3509dfb4 fixes the issue :)

**Comment by binji** [comment]
Hm, weird. I wonder if it has to do with using `SDLWINDOW_HIGHDPI` in the window creation call.

**Comment by binji** [comment]
Yep, seems to. I tested on my Mac laptop, [this change](https://github.com/binji/binjgb/commit/1ac328bf5533ea2ac526da7c85fe1c8b7da4650d) should fix it!

**Comment by aconbere** [comment]
@binji Works! And now in debug mode I actually get all the debug tools :)

The funny thing about all of this is I ended up here following a lead in a reddit comment (from you) saying that your emulator had good instruction logging. I'm trying to get a personal emulator over the finish line where there's a bug in some instruction and I just can't quite figure out which one. Whatever it is prevents a successful from of blargg's test suite and so I'm a bit stumped.

What I've got now is a way to run my emulator against a file that tracks known good register states and will compare its incremental state to those and log when it differs so that I can dig into exactly where it goes off the rails.

Anyway, appreciate how quickly you've been responding here, and I've nearly got my state file built from your emulator :)
